index.js

```diff
@@ -292,27 +306,29 @@ module.exports = class Filter extends Plugin {
     });
   }
 
-  async _handleFile(relativePath, srcDir, destDir, entry, outputPath, forceInvalidation, isChange, stats) {
+  async _handleFile(relativePath, entry, outputPath, forceInvalidation, isChange, stats) {
     stats.handleFile++;
 
     let handleFileStart = process.hrtime();
+    let destDir = this.outputPath + '/' + outputPath;
     try {
       let result;
-      let srcPath = srcDir + '/' + relativePath;
+      let srcDir = this.inputPaths[0]; // keeping this line to maintain the signature of the fn.
```

Comment:
Using `this.inputPaths[0]` here is breaking encapsulation, post `this.input` and `this.output` migration it should be possible to use only the new APIs, what features are missing or different approaches are required to make this statement true.